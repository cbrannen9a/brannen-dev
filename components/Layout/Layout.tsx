import { FC } from "react";
import Alert from "../Alert";
import Footer from "../Footer";
import Meta from "../Meta";

const Layout: FC<Props> = ({ preview, children }) => (
  <>
    <Meta />
    <div className="min-h-screen">
      <Alert preview={preview} />
      <main>{children}</main>
    </div>
    <Footer />
  </>
);

interface Props {
  preview: any;
}

export default Layout;
