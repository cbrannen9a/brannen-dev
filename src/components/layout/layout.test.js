import React from 'react';
import renderer from 'react-test-renderer';

import Layout from './layout';

describe('Layout', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<Layout><div>Test Child</div></Layout>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});