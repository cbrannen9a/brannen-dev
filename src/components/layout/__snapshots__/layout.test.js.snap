// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div>
  <main>
    <div>
      Test Child
    </div>
  </main>
  <footer>
    Â© 
    2019
    , Built with
     
    <a
      href="https://www.gatsbyjs.org"
    >
      Gatsby
    </a>
  </footer>
</div>
`;
